velox/expression/CastExpr-inl.h

```diff
@@ -286,6 +286,23 @@ void CastExpr::applyCastKernel(
       return;
     }
 
+    if constexpr (
+        (FromKind == TypeKind::DOUBLE || FromKind == TypeKind::REAL) &&
+        ToKind == TypeKind::TIMESTAMP) {
+      const auto castResult =
+          hooks_->castDoubleToTimestamp(static_cast<double>(inputRowValue));
+      if (castResult.hasError()) {
+        if (castResult.error().isUserError()) {
```

Comment:
The error handling according to type seems to be Spark-specific. I'm wondering if it's possible to wrap it inside the cast hook.

Reply:
Since we have to return null for Nan and Inf, Looks Expected can't support return null value, so here I update to use `Expected<std::optional<Timestamp>>` to wrap the logic inside Spark implementation.